src/devices/lacrosse_TX141TH_Bv2.c

```diff
@@ -0,0 +1,230 @@
+ * the true data. Therefore, in the fisrt part of the callback routine the mode is determined
+ * for the first 4 bytes of the data compressed into a single 32-bit integer. Since the packet
+ * count is small, no sophisticated mode algorithm is necessary; a simple array of <data,count>
+ * structures is sufficient. The added bonus is that relative count enables us to determine
+ * the quality of radio transmission.
+ *
+ * Copyright (C) 2017 Robert Fraczkiewicz   (aromring@gmail.com)
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ */
+#include "data.h"
+#include "rtl_433.h"
+#include "util.h"
+
+#define LACROSSE_TX141TH_BITLEN 40
+#define LACROSSE_TX141TH_BYTELEN 5  // = LACROSSE_TX141TH_BITLEN / 8
+#define LACROSSE_TX141TH_PACKETCOUNT 12
+
+typedef struct {
+    int32_t	data;	// First 4 data bytes compressed into 32-bit integer
+    int8_t count;	// Count
+} data_and_count;
+
+static int lacrosse_tx141th_bv2_callback(bitbuffer_t *bitbuffer) {
+	bitrow_t *bb = bitbuffer->bb;
+	data_t *data;
+	char time_str[LOCAL_TIME_BUFLEN];
+    local_time_str(0, time_str);
+	int i,j,k,nbytes,npacket,kmax;
+    uint8_t id=0,status=0,battery_low=0,test=0,humidity=0,maxcount;
+    uint16_t temp_raw=0;
+    float temp_c=0.0;
+    data_and_count dnc[LACROSSE_TX141TH_PACKETCOUNT];
+    memset(dnc,0,LACROSSE_TX141TH_PACKETCOUNT*sizeof(data_and_count));
+    
+    if (debug_output) {
+        bitbuffer_print(bitbuffer);
+    }
+    
+    npacket=0; // Number of unique packets
+    for(i=0; i<BITBUF_ROWS; ++i) {
+        j=bitbuffer->bits_per_row[i];
+        if(j>=LACROSSE_TX141TH_BITLEN) {
+            nbytes=j/8;
+            for(j=0;j<nbytes;j+=LACROSSE_TX141TH_BYTELEN) {
+                uint32_t *d=(uint32_t *)(bb[i]+j);
+                uint8_t not_found=1;
+                for(k=0;k<npacket;++k) {
+                    if(*d==dnc[k].data) {
+                        ++(dnc[k].count);
+                        not_found=0;
+                        break;
+                    }
+                }
+                if(not_found) {
+                    dnc[npacket].data=*d;
+                    dnc[npacket].count=1;
+                    if(npacket+1<LACROSSE_TX141TH_PACKETCOUNT) ++npacket;
+                }
+            }
+        }
+    }
+ 
+    if (debug_output) {
+        fprintf(stderr, "%d unique packet(s)\n", npacket);
+        for(k=0;k<npacket;++k) {
+            fprintf(stderr, "%08x \t %d \n", dnc[k].data,dnc[k].count);
+        }
+    }
+
+    // Find the most frequent data packet, if necessary
+    kmax=0;
+    maxcount=0;
+    if(npacket>1) {
+        for(k=0;k<npacket;++k) {
+            if(dnc[k].count>maxcount) {
+                maxcount=dnc[k].count;
+                kmax=k;
+            }
+        }
+    }
```

Comment:
Row 130 to 171 should be removable if we can figure out the crc. Can you post a few bitstream dumps ?